### ISSUE TITLE
Adding a second Document object using the admin site

### ISSUE THREAD

**Issue by philippeowagner** [danielquinn/paperless#195]
ends in an `IntegrityError at /admin/documents/document/add/` exception.
What is the accepted/suggested way to fix this @danielquinn ? I have to avoid this to be able to add Markdown documents to paperless.

**Comment by philippeowagner** [comment]
.. and when trying to export the documents, I got 

```
Exporting: /mnt/volume.........-blah-test-blah.
Exception in thread Thread-9:
Traceback (most recent call last):
  File "/usr/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/mnt/volume-....paperless/lib/python3.4/site-packages/gnupg.py", line 147, in _copy_data
    data = instream.read(1024)
AttributeError: 'str' object has no attribute 'read'
```

**Comment by philippeowagner** [comment]
My suggestion is the following: Overwriting the `save` method of the Document model and use a fix string including the id of the object for the checksum. This would fix the first part of the issue.

```
    def save(self, *args, **kwargs):
        if not self.id:
            import hashlib
            self.checksum = hashlib.md5(("paperless-document-{}".format(self.id)).encode('utf-8')).hexdigest()
        super(Document, self).save(*args, **kwargs)
```
This allows to create document objects w/o files.

**Comment by danielquinn** [comment]
I'm confused.  Are you trying to create a `document` object without a document?  The admin may have a form for adding documents, but it's not actually meant to work ;-)

The problem is that Django assumes that everything attached to an object lives in the database and when I was originally writing Paperless, I didn't want to include file information with it (I don't remember why now) so the admin simply isn't capable of handing document creation at present.  Instead, you have to go the way of placing your document into the consumption directory and letting the consumer do its thing.  In fact, I just patched `documents.admin` to disable adding permission so this doesn't surprise anyone else in the future.

The proper fix for this would be to fix the document model to make use of Django's `FileField` both for the `document` value and for the `thumbnail` but that's no small job at this stage, in large part because I'd have to include some sort of elaborate migration script for all the users out there.  I'd probably want to combine that fix with using [cryptography](https://pypi.python.org/pypi/cryptography) instead of GnuPG too.

So TL;DR:

* All document additions must currently go through the consumer
* I'm not against a proper fix, but it's not a small job
